documentation/disaster-recovery-testing.md

```diff
@@ -0,0 +1,208 @@
+- Recreate the lost postgres database instance
+- Restore data from nightly backup stored in Azure.
+
+The point-in-time and snapshot backups created by the Azure Postgres service may not be available if it's been deleted, or if there is an Azure region issue.
+
+### Delete the postgres database instance
+
+Note that you must have a previously created backup on azure storage before starting this step. If not, create one now before continuing.
+
+- Delete the existing postgres database
+    - manually delete via UI https://portal.azure.com/#browse/Microsoft.DBforPostgreSQL%2FflexibleServers
+- confirm it's deleted
+
+### Enable maintenance mode
+
+Run the enable maintenance workflow for the service and environment affected
+
+### Recreate the lost postgres database instance
+
+Run the deploy workflow to recreate the missing postgres database (and any other missing resources).
+
+As the maintenance page has been implemented, you will need to
+- create a branch from main
+- update the env tfvars.json as per https://github.com/DFE-Digital/teacher-services-cloud/blob/main/documentation/maintenance-page.md#configure-terraform-to-keep-deploying-the-application
+- push the branch to github
+- run the deploy workflow using your branch for the env being tested
+
+The web app may fail to start due to the postgres db missing data. It needs to be running so that the restore workflow can connect to the database using konduit. If it is failing, update your branch
+- override the start command to remove any db commands or a dummy command (e.g. sleep 999999).
+- or remove startup/liveness probes
+- or any other method that enables the web app to start
+- push the branch and rerun deploy workflow as above
+
+### Restore the data from previous backup in Azure storage
+
+Run the restore from backup workflow
+
+### Redeploy app if required
+
+If the start command was updated previously
+- put back the original start command in your PR
+- push to github and rerun deploy
+
+### Validate app
+
+Confirm the app is working and is using the restored data
+
+
+## Scenario 2. Loss of data
+
+In the case of data loss or corruption, we need to recover the data as soon as possible in order to resume normal service.
+
+The application's database is an Azure flexible postgres instance. This provides a point-in-time restore (PTR) ability with the resolution of 1 second, available between 5min and 7days. PTR allows you to restore the source server to a point-in-time on a new copy of the server, not to the source server itself.
+
+The objectives of this test are:
+- Create a separate new postgres database instance
+- Restore data from the current live database to the new postgres database instance from a particular point in time
+- Update data into the source database from the new PTR server
+
+### Delete data from the postgres database instance
+
+Make a note of the time this step is being started as the restore point must be before you delete any data
+
+- Delete a table manually
```

Comment:
It should be visible and verifiable in the app: the app may crash, show errors, or miss data. The point is to verify easily and visually after the restore that it was successful

Reply:
Made this clearer in the documentation